test/decl/protocol/req/missing_conformance.swift

```diff
@@ -107,7 +107,7 @@ protocol P12 {
 extension Int : P11 {}
 struct S3 : P12 { // expected-error {{type 'S3' does not conform to protocol 'P12'}}
     func bar() -> P11 { return 0 }
-    // expected-note@-1 {{candidate can not infer 'A' = 'P11' because 'P11' is not a nominal type and so can't conform to 'P11'}}
+    // expected-note@-1 {{candidate can not infer 'A' = 'P11' because 'P11' is not a struct, class or enum type and so can't conform to 'P11'}}
```

Comment:
This is better, but it's still potentially confusing to users.

It still doesn't answer the question why `P11` doesn't self conform; for this, it would be helpful to point users to the other educational note on existential types.

Moreover, users may not be familiar at all with the notion that the existential type is distinct from the corresponding protocol at all. Therefore, it would be helpful to explain (instead of what `P11` is _not_) what `P11` _is_ (that is, a protocol used as a type). For me, a message such as the following would be more on the nose:

"candidate cannot infer 'A' = 'P11' because 'P11' is a protocol used as a type, which can't conform to 'P11' or any other protocol"

Reply:
Good point, I think what you propose sounds a lot better! I’ll tweak the text.